File: backend/app/schemas/daily_brief.py
from datetime import date, datetime
from uuid import UUID

from pydantic import BaseModel, Field

from app.schemas.base import ORMModel


class BriefProgressItem(BaseModel):
    type: str
    title: str
    detail: str | None = None


class BriefContext(BaseModel):
    what_matters: list[str] = Field(default_factory=list)
    recent_progress: list[BriefProgressItem] = Field(default_factory=list)
    focus_topics: list[str] = Field(default_factory=list)
    communication_style: str | None = None


class DailyBriefOut(ORMModel):
    id: UUID
    user_id: UUID
    brief_date: date
    summary: str
    open_loops: list[str] = Field(default_factory=list)
    next_step: str
    context: BriefContext
    created_at: datetime

File: frontend-v2/types/daily-brief.ts
export type BriefProgressItem = {
  type: "project" | "conversation" | "task" | string;
  title: string;
  detail: string | null;
};

export type DailyBrief = {
  id: string;
  user_id: string;
  brief_date: string;
  summary: string;
  open_loops: string[];
  next_step: string;
  context: {
    what_matters: string[];
    recent_progress: BriefProgressItem[];
    focus_topics: string[];
    communication_style: string | null;
  };
  created_at: string;
};
